spec/controllers/locations_controller_spec.rb

```diff
@@ -78,4 +81,46 @@
       end
     end
   end
+
+  describe "POST /update_location" do
+    context "when sign_in user is super_admin" do
+      before do
+        sign_in super_admin
+      end
+
+      context "when location's organisation belongs to signed_in user organisation" do
+        it "update location and redirect to ips_path " do
+          put :update_location, params: { location_id: location.id, location: { organisation_id: user.organisations.first.id, address: location.address, postcode: location.postcode } }
+          expect(response).to redirect_to(ips_path)
+        end
+      end
+
+      context "when location's organisation doesn't belong to signed_in user organisation" do
+        it "render exit page with flash alert" do
+          put :update_location, params: { location_id: location.id, location: { organisation_id: other_organisation.id, address: location.address, postcode: location.postcode } }
+          expect(response).to redirect_to(ips_path)
+        end
+      end
+    end
+
+    context "when sign_in user is not a super_admin" do
+      before do
+        sign_in user
+      end
+      context "when location's organisation belongs to signed_in user organisation" do
+        it "update location and redirect to ips_path " do
+          put :update_location, params: { location_id: location.id, location: { organisation_id: user.organisations.first.id, address: location.address, postcode: location.postcode } }
+          expect(response).to redirect_to(ips_path)
+        end
+      end
+
+      context "when location's organisation doesn't belong to signed_in user organisation" do
+        it "render exit page with flash alert" do
+          put :update_location, params: { location_id: location.id, location: { organisation_id: other_organisation.id, address: location.address, postcode: location.postcode } }
+          expect(flash[:alert]).to match(/You can't update the location to an organisation you're not a member of/)
+          expect(response).not_to redirect_to(ips_path)
+        end
+      end
+    end
+  end
```

Comment:
There are a number of other constraints we need to take into account when moving locations:

1. The user needs to have the right permissions on both the source and destination organisation.
2. The destination organisation must have at least 2 administrators